### PR TITLE
Files needed by nvdi code review partner

### PR DESCRIPTION
**Pull request by rami8797**
# Pull Request Template

## CI Checks
- [ ] The notebook runs from start to finish on all operating systems:
   - [ ] Mac
   - [ ] Windows
   - [ ] Linux 
### Reproducibility
- [ ] Are the data downloaded in the code
- [ ] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [ ] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [ ] Are function names expressive (the name describes what the function does)?
- [ ] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [ ] Are variable names throughout the code, expressive?
- [ ] Suggest changes if not, highlight what was done well
- [ ] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [ ] Are segments of code repeated in the file or is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [ ] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

**Review by AreteY — COMMENTED** [review]
Hi @rami8797 ! Just a few comments for your code! I'll submit an issue shortly! :)

File: michaels_rachel_ndvi.py [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+    bound = os.path.join(path, "vector", "HARV-crop.shp")
+    # Open the shapefile
+    harv_boundary = gpd.read_file(bound)
+
+    # Create a nested for loop to be able to work with each .tif file (band)
+    # in the scene, again this is necessary when working with multiple scenes
+    for tif in scene_path:
+        # Get site and date info from the scene directory path
+        site_info = extract_sitename_date(tif, [22, 26], [50, 58])
+        # Sort the bands using glob so that they are in the right order
```
[the remark this answers is not in the document]
Comment (rami8797) [reply michaels_rachel_ndvi.py:358]: I agree, @AreteY this comment is probably not needed.

File: michaels_rachel_ndvi.py [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+
+    Returns
+    -----------
+    list : list of site names and datetime information
+    """
+    # Create an empty list to append sitename and date information
+    site_name_date_list = []
+
+    # Assign datetime location to an object
+    date_location = directory_path[datetime_location[0]: datetime_location[1]]
```
[the remark this answers is not in the document]
Comment (rami8797) [reply michaels_rachel_ndvi.py:279]: Ah, yep, thanks for pointing that out.

File: michaels_rachel_ndvi.py [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+                          sitename_location,
+                          datetime_location):
+    """Extract sitename and datetime from directory path name.
+
+    Parameters
+    -----------
+    directory_path : string
+        A path to the directory name
+    sitename_location : index list
+        Index of sitename location in directory path name
```
[the remark this answers is not in the document]
Comment (rami8797) [reply michaels_rachel_ndvi.py:266]: Your suggestion is more descriptive and clearer! I'll definitely use that description in the future.

File: michaels_rachel_ndvi.py [hunk 4]
```diff
@@ -0,0 +1,729 @@
...
+# Create path to HARV data
+harv_path = glob(os.path.join("ndvi-automation", "sites", "HARV"))
+
+# Open and clean all HARV bands
+harv_scene_info = []
+
+# Create a loop to establish the scene directory path
+# glob is not necessary here, however it is for the larger workflow, which is
+# what is being demonstrated here
+for path in harv_path:
```
[the remark this answers is not in the document]
Comment (rami8797) [reply michaels_rachel_ndvi.py:344]: You're absolutely right, thanks!

**Pull request closed without merging**